xonsh/completer.py

```diff
@@ -171,14 +171,15 @@ def generate_completions(
 
             items = []
             for comp in res:
-                items.append(comp)
-                yield Completer._format_completion(
+                comp = Completer._format_completion(
                     comp,
                     completion_context,
                     completing_contextual_command,
                     lprefix or 0,
                     custom_lprefix,
                 )
+                items.append(comp)
+                yield comp
```

Comment:
this is just a refactoring ?

Reply:
No, the completions that were traced weren't the ones actually being returned because of `_format_exception`